### PR TITLE
Test support for ActorSystem.shutdown and ShutdownHooks

### PR DESCRIPTION
**Pull request by kolloch**
Based on PR #2891

### CODE REVIEW

File: src/test/scala/mesosphere/marathon/upgrade/TaskStartActorTest.scala [hunk 1]
```diff
@@ -48,6 +50,7 @@ class TaskStartActorTest
   override protected def afterAll(): Unit = {
     super.afterAll()
     system.shutdown()
+    system.awaitTermination()
```
Comment (aquamatthias) [inline src/test/scala/mesosphere/marathon/upgrade/TaskStartActorTest.scala]: does it make sense to add this functionality to the MarathonActorSupport trait?

**Comment by aquamatthias** [comment]
LGTM! - please rebase

File: src/test/scala/mesosphere/marathon/upgrade/TaskStartActorTest.scala (hunk 1 above)
Comment (kolloch) [inline src/test/scala/mesosphere/marathon/upgrade/TaskStartActorTest.scala]: It is included. I forgot to remove this. Thanks!

**Comment by kolloch** [comment]
rebased.

**Comment by aquamatthias** [comment]
Thanks!

**Pull request merged**